### PR TITLE
feat(components): [table] expose scroll event

### PR DESCRIPTION
**Pull request by blesstosam**
Sometimes we want to handle something when table is scrolling，but table component does not expose the `scroll` event. I also noticed table-v2 exposed this event, so why not table?

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Comment by btea** [comment]
@blesstosam Hi, thanks for your contribution, but I don't think this one is necessary. If you want to monitor element scrolling, maybe you can directly get the corresponding element and bind the event.

**Comment by blesstosam** [comment]
> @blesstosam Hi, thanks for your contribution, but I don't think this one is necessary. If you want to monitor element scrolling, maybe you can directly get the corresponding element and bind the event.

Sometimes we want to handle something when table is scrolling. I think directly listen to this event is much more efficient than listen to a specific a html element. btw, I also noticed table-v2 exposed this event, so why not table?

**Comment by codesyofo** [comment]
需要该事件，多个表格滚动联动的场景 @btea

**Comment by Spike-Leung** [comment]
> 需要该事件，多个表格滚动联动的场景 @btea

same too.

**Comment by btea** [comment]
@blesstosam Can you resolve the code conflict?

File: docs/en-US/component/table.md [hunk 1]
```diff
@@ -312,6 +312,7 @@ table/table-layout
 | current-change     | triggers when current row changes                                                                                                            | ^[Function]`(currentRow: any, oldCurrentRow: any) => void`                                   |
 | header-dragend     | triggers after changing a column's width by dragging the column header's border                                                              | ^[Function]`(newWidth: number, oldWidth: number, column: any, event: MouseEvent) => void`    |
 | expand-change      | triggers when user expands or collapses a row (for expandable table, second param is expandedRows; for tree Table, second param is expanded) | ^[Function]`(row: any, expandedRows: any[]) => void & (row: any, expanded: boolean) => void` |
+| scroll             | Invoked after scrolled                                                                                                                       | Object                                                                                       |
```
Comment (btea) [inline docs/en-US/component/table.md:315]:
```suggestion
| scroll ^(2.9.0)    | Invoked after scrolled                                                                                                                       | Object                                                                                       |
```

Comment (btea) [inline docs/en-US/component/table.md:315]: The scroll type should be a function, and the parameter accepted by the function can be directly the object dispatched by the `scrollbar` component `{scrollTop: number, scrollLeft: number }`.

https://github.com/element-plus/element-plus/blob/84fb6933c39a51245911ea99245171477579ae73/packages/components/scrollbar/src/scrollbar.vue#L95-L98

**Review by btea — APPROVED** [review]
Thanks!

**Pull request merged**